src/test/compile-fail/private-in-public-warn.rs

```diff
@@ -287,5 +288,15 @@ mod aliases_params {
     type Result<T> = ::std::result::Result<T, Priv>;
 }
 
+mod foo {
+    struct Priv;
+    pub mod bar {
+        use super::Priv;
+        pub(super) fn f(_: Priv) {} // ok
+        pub(crate) fn g(_: Priv) {} //~ WARN private
```

Comment:
Maybe private-in-public visitor one more temporary flag to report everything as errors if `pub(restricted)` is involved (not necessarily precisely).